src/main/java/frc/robot/subsystems/PivotSubsystem.java

```diff
@@ -1,84 +1,111 @@
-
-    public PivotSubsystem(CANSparkMax pivotMotor) {
-        this.pivotMotor = pivotMotor;
-        
-        this.pivotEncoder = pivotMotor.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle);
-        this.pivotEncoder.setZeroOffset(ArmConstants.pivotOffset);
-
-        pivotPidController.enableContinuousInput(0, 1);
-
-        targetPos = pivotEncoder.getPosition();
-    }
-
-    public void setPivotMotorVoltage(double speed) {
-        
-        pivotMotor.setVoltage(speed);
-    }
-
-    public void setPivotMotor(double speed) {
-        pivotMotor.set(speed);
-    }
-
-    public void setPivotPosition(double pos) {
-        // System.out.println(ntPivotPosition.getDouble(0));
-        pivotMotor.setVoltage(armFeedforward.calculate(pos, 0) + pivotPidController.calculate(ntPivotPosition.getDouble(0), pos));
-    }
-
-    @Override
-    public void periodic() {
-        updatePivotPosition();
-        this.updateArmFeedForward(ArmConstants.startingPivotG + ArmConstants.pivotGPerTelescopeMeter*ntTelescopeLength.getDouble(0));
-    }
-
-    public void updatePivotPosition() {
-        ntPivotPosition.setDouble(pivotEncoder.getPosition());
-    }
-
-    public double getPivotPosition() {
-        return ntPivotPosition.getDouble(0);
-    }
-
-    public void updateArmFeedForward(double newG) {
-        armFeedforward = new ArmFeedforward(0, newG, 0);
-    }
-
-    public boolean getPivotInRange(double setpoint, double buffer){
-        return Math.abs(getPivotPosition() - setpoint) < buffer || Math.abs(getPivotPosition() - setpoint) > 1 - buffer;
-    }
-
-    public void setTargetPivot(double targetPos) {
-        this.targetPos = targetPos;
-    }
-
-    public double getTargetPivot(){
-        return targetPos;
-    }
+  private final CANSparkMax pivotMotor;
+  private final SparkMaxAbsoluteEncoder pivotEncoder;
+
+  private final TalonFXSensorCollection telescopeSensor;
+
+  private final NetworkTableInstance ntInstance = NetworkTableInstance.getDefault();
+  private final NetworkTable table = ntInstance.getTable("/components/arm");
+
+  private final NetworkTableEntry ntPivotPosition = table.getEntry("pivotPosition");
+
+  private final PIDController pivotPidController;
+
+  private double targetPos;
+
+  public PivotSubsystem(CANSparkMax pivotMotor, TalonFXSensorCollection telescopeSensor) {
+    this.pivotMotor = pivotMotor;
+    this.telescopeSensor = telescopeSensor;
+
+    this.pivotEncoder = pivotMotor.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle);
+    this.pivotEncoder.setZeroOffset(ArmConstants.pivotOffset);
+
+    this.pivotPidController = new PIDController(19, 0, 3);
+    this.pivotPidController.setTolerance(3, 1);
+    this.pivotPidController.enableContinuousInput(0, 1);
+
+    targetPos = pivotEncoder.getPosition();
+
+    setDefaultCommand(moveToTargetContinuously());
+  }
+
+  @Override
+  public void periodic() {
+    ntPivotPosition.setDouble(pivotEncoder.getPosition());
```

Comment:
`periodic` is only used for updating internal state (like network tables), not external state (like motor values)